pkg/controller/atlasdeployment/serverless_private_endpoint.go

```diff
@@ -0,0 +1,356 @@
+package atlasdeployment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/stringutil"
+
+	"go.uber.org/zap"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/provider"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+const (
+	SPEStatusDeleting = "DELETING"
+
+	SPEStatusReserved             = "RESERVED"              // 1
+	SPEStatusReservationRequested = "RESERVATION_REQUESTED" // 1
+
+	SPEStatusAvailable  = "AVAILABLE"  // 2
+	SPEStatusInitiating = "INITIATING" // 2
+	SPEStatusFailed     = "FAILED"     // 2
+)
+
+func ensureServerlessPrivateEndpoints(service *workflow.Context, groupID string, deploymentSpec *mdbv1.ServerlessSpec, deploymentName string) workflow.Result {
+	if deploymentSpec == nil {
+		return workflow.Terminate(workflow.ServerlessPrivateEndpointReady, "deployment spec is empty")
+	}
+	if result := IsClusterSupportServerlessPE(deploymentSpec); !result.IsOk() {
+		return result
+	}
+
+	if deploymentSpec.PrivateEndpoints == nil {
+		service.UnsetCondition(status.ServerlessPrivateEndpointReadyType)
+		service.EnsureStatusOption(status.AtlasDeploymentServerlessPrivateEndpointOption(nil))
```

Comment:
This naming `AtlasDeploymentServerlessPrivateEndpointOption` 😅
Maybe we can cut at least something?

Reply:
Done